src/test/java/io/confluent/idesidecar/restapi/resources/ConnectionsResourceTest.java

```diff
@@ -1280,7 +1276,99 @@ record TestInput(
                 "bootstrap_servers": "localhost:9092"
               },
               "schema_registry": {
-                "uri": "http://localhost:8081"
+                "uri": "http://localhost:8081",
+                "ssl": { "enabled": true }
+              }
+            }
+            """
+        ),
+        new TestInput(
+            "Direct spec is valid with Kafka and verify server certificate hostname",
+            """
+            {
+              "name": "Some connection name",
+              "type": "DIRECT",
+              "kafka_cluster": {
+                "bootstrap_servers": "localhost:9092",
+                "ssl": { "enabled": true, "verify_hostname": true}
+              }
+            }
+            """
+        ),
+        new TestInput(
+            "Direct spec is valid with Kafka and don't verify server certificate hostname",
+            """
+            {
+              "name": "Some connection name",
+              "type": "DIRECT",
+              "kafka_cluster": {
+                "bootstrap_servers": "localhost:9092",
+                "ssl": { "enabled": true, "verify_hostname": false}
+              }
+            }
+            """
+        ),
+        new TestInput(
+            "Direct spec is valid with Kafka and SR over TLS",
+            """
+            {
+              "name": "Some connection name",
+              "type": "DIRECT",
+              "kafka_cluster": {
+                "bootstrap_servers": "localhost:9092",
+                "ssl": {
+                  "enabled": true,
+                  "truststore": {
+                    "path": "/path/to/truststore.jks",
+                    "password": "truststore-password"
+                  }
+                }
+              },
+              "schema_registry": {
+                "uri": "https://localhost:8081"
+              }
+            }
+            """
+        ),
+        new TestInput(
+            "Direct spec is valid over SSL with truststore path only",
+            """
+            {
+              "name": "Connection 1",
+              "type": "DIRECT",
+              "kafka_cluster": {
+                "bootstrap_servers": "localhost:9092"
+              },
+              "ssl": {
+                "truststore": {
+                  "path": "/path/to/truststore.jks"
+                }
+              }
```

Comment:
Should this be under `kafka_cluster`?

Reply:
Yes, thanks for catching that. Seems we don't complain about unknown fields in the JSON payload.